iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -149,13 +162,15 @@ inline optional<T>::~optional() noexcept
 template <typename T>
 template <typename U>
 inline typename std::enable_if<!std::is_same<U, optional<T>&>::value, optional<T>>::type&
-optional<T>::operator=(U&& newValue) noexcept
+optional<T>::operator=(U&& newValue) & noexcept
 {
     if (m_hasValue)
     {
 /// @todo broken msvc compiler, see:
 ///  https://developercommunity.visualstudio.com/content/problem/858688/stdforward-none-of-these-2-overloads-could-convert.html
 /// remove this as soon as it is fixed;
+// AXIVION DISABLE STYLE AutosarC++19_03-A16.0.1 : this macro is used to prevent a large amount
+// of boilerplate code
```

Comment:
Instead of suppressing the warning I suggest to remove the following three lines and check if the msvc compiler has fixed the issue in the meantime.
When the CI is green than off it goes.

Reply:
@elfenpiff let me know about this. thanks.